src/ServiceControl.Audit/Auditing/AuditEnricherContext.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace ServiceControl.Audit.Auditing
+{
+    using NServiceBus;
+    using System.Collections.Generic;
+
+    class AuditEnricherContext
+    {
+        public AuditEnricherContext(IReadOnlyDictionary<string, string> headers, IList<IEvent> outgoingEvents, IDictionary<string, object> metadata)
+        {
+            Headers = headers;
+            this.outgoingEvents = outgoingEvents;
+            Metadata = metadata;
+        }
+
+        public IReadOnlyDictionary<string, string> Headers { get; }
+
+        public IDictionary<string, object> Metadata { get; }
+
+        public void Emit(IEvent @event)
```

Comment:
Would `Add`, `AddForPublish`, `AddForEmit` or something similar be a better name?

Reply:
I like `AddForPublish` since we are still tied to a publish operation in the end. Renamed...